beacon_node/store/src/hot_cold_store.rs

```diff
@@ -436,6 +441,22 @@ impl<E: EthSpec> HotColdDB<E, LevelDB<E>, LevelDB<E>> {
 }
 
 impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold> {
+    fn cold_storage_strategy(&self, slot: Slot) -> Result<StorageStrategy, Error> {
+        // The start slot for the freezer HDiff is always 0
+        Ok(self.hierarchy.storage_strategy(slot, Slot::new(0))?)
+    }
+
+    pub fn hot_storage_strategy(&self, slot: Slot) -> Result<StorageStrategy, Error> {
+        Ok(self
+            .hierarchy_hot
+            .storage_strategy(slot, self.hot_hdiff_start_slot())?)
+    }
+
+    pub fn hot_hdiff_start_slot(&self) -> Slot {
+        // TODO(hdiff): read the start slot from somewhere
+        todo!();
```

Comment:
I was thinking maybe we could recycle the `anchor_slot` for use here, as it is _fairly useless_ and not used for much. The only load-bearing use of the `anchor_slot` that I can find is in `try_prune_execution_payloads` where it is used to halt the pruning process. However, this is not necessary as we could either:

1. Keep iterating back to Bellatrix, or
2. Stop iterating back once we find a payload that is missing, or
3. Use the `oldest_block_slot` to determine when to stop iterating back. This is my preferred option, as it is also compatible with storing execution payloads older than the `anchor_slot`, i.e.

- https://github.com/sigp/lighthouse/issues/6510

Reply:
Conclusion: we can recycle the `anchor_slot` field and set it to the epoch-aligned slot of the snapshot state (_not_ the slot of the checkpoint block, which might be older due to skipped slots).

Reply:
Fixed on 5ecf9dbf9ced9e8dd9fcc8f3e3aa3fb2ce12f5b1